source/nodejs/adaptivecards/src/card-elements.ts

```diff
@@ -2452,7 +2460,234 @@ export class MediaSource extends SerializableObject {
+                    const index = HTML5MediaPlayer.supportedMediaTypes.indexOf(mimeComponents[0]);
+
+                    if (index >= 0) {
+                        this._selectedMediaType = HTML5MediaPlayer.supportedMediaTypes[index];
+                    }
+                }
+                if (mimeComponents[0] === this._selectedMediaType) {
+                    this._selectedSources.push(source);
+                }
+            }
+        }
+    }
+
+    static readonly supportedMediaTypes = [ "audio", "video" ];
+
+    constructor(readonly owner: Media) {
+        super();
+
+        this.processSources();
+    }
+
+    canPlay(): boolean {
+        return this._selectedSources.length > 0;
+    }
+
+    async fetchVideoDetails() {
+        // Nothing to fetch for the HTML5 media player
+    }
+
+    render(): HTMLElement {
+        if (this._selectedMediaType === "video") {
+            this._mediaElement = document.createElement("video");
+        } else {
+            this._mediaElement = document.createElement("audio");
+        }
+
+        this._mediaElement.setAttribute(
+            "aria-label",
+            this.owner.altText ? this.owner.altText : Strings.defaults.mediaPlayerAriaLabel()
+        );
+        this._mediaElement.setAttribute("webkit-playsinline", "");
+        this._mediaElement.setAttribute("playsinline", "");
+        this._mediaElement.autoplay = true;
+        this._mediaElement.controls = true;
+
+        if (Utils.isMobileOS()) {
+            this._mediaElement.muted = true;
+        }
+
+        this._mediaElement.preload = "none";
+        this._mediaElement.style.width = "100%";
+
+        for (const source of this.owner.sources) {
+            const renderedSource = source.render();
+
+            Utils.appendChild(this._mediaElement, renderedSource);
+        }
+
+        return this._mediaElement;
+    }
+
+    play() {
+        if (this._mediaElement) {
+            this._mediaElement.play();
+        }
+    }
+
+    get selectedMediaType(): string | undefined {
+        return this._selectedMediaType;
+    }
+}
+
+export abstract class CustomMediaPlayer extends MediaPlayer {
+    constructor(matches: RegExpExecArray) {
+        super();
+    }
+}
+
+export abstract class IFrameMediaMediaPlayer extends CustomMediaPlayer {
+    private _videoId?: string;
+
+    constructor(matches: RegExpExecArray) {
+        super(matches);
+
+        if (matches.length === 2) {
+            this._videoId = matches[1];
+        }
+    }
+
+    abstract getEmbedVideoUrl(): string;
+
+    canPlay(): boolean {
+        return this._videoId !== undefined;
+    }
+
+    render(): HTMLElement {
+        let container = document.createElement("div");
+        container.style.position = "relative";
+        container.style.width = "100%";
+        container.style.height = "0";
+        container.style.paddingBottom = "56.25%";
+
+        let iFrame = document.createElement("iframe");
```

Comment:
I believe we need to [specify the `title` property for accessibility](https://dequeuniversity.com/tips/provide-iframe-titles) purposes....

Reply:
In my tests, `title` is never read, which is why I didn't include it. It's easy enough to add it if you insist. We just need to make it localizable.

Reply:
Fixed anyway.